test/govspeak_attachment_link_test.rb

```diff
@@ -22,4 +22,19 @@ def render_govspeak(govspeak, attachments = [])
     assert_match(/<span class="gem-c-attachment-link">/, rendered)
     assert_match(%r{href="http://example.com/attachment.pdf"}, rendered)
   end
+
+  test "renders an attachment link inside a paragraph" do
+    attachment = {
+      id: "attachment.pdf",
+      url: "http://example.com/attachment.pdf",
+      title: "Attachment Title",
+    }
+
+    rendered = render_govspeak("[AttachmentLink:attachment.pdf] test", [attachment])
+    root = Nokogiri::HTML.fragment(rendered).css(":root").first
+
+    assert(root.name, "p")
+    assert(root.css("p").size, 0)
+    assert_match(/Attachment Title\s*test/, root.text)
```

Comment:
Oh dear, this isn't as simple as I thought it might be. I was hoping we could do some regex piece, but I can see how you got here. Up to you if we keep it though ;-).

Reply:
Yup regex was fraught with complications. I'll just merge this.